packages/react/src/components/timeInput/TimeInput.tsx

```diff
@@ -0,0 +1,317 @@
+ */
+const incrementNumber = (min: number, max: number, current: number, modifier: number) => {
+  return Math.max(Math.min(current + modifier, max), min);
+};
+
+/**
+ * Returns the default hour/minute values from a defaultValue string
+ */
+const getDefaultValues = (defaultValue?: string): string[] | null => {
+  const defaultValueString = `${defaultValue}`;
+  if (defaultValue && defaultValueString.length > 0) {
+    if (defaultValueString.match(/^\d{2}:\d{2}$/)) {
+      return defaultValueString.split(':');
+    }
+    // eslint-disable-next-line no-console
+    console.warn('Invalid default value for TimeInput. The default value must be in hh:mm format');
+  }
+  return null;
+};
+
+const isShortNumericString = (inputValue: string): boolean => inputValue.match(/^(\d{1,2})?$/) !== null;
+
+export const TimeInput = React.forwardRef<HTMLInputElement, TextInputProps>(
+  (
+    {
+      className = '',
+      disabled = false,
+      defaultValue,
+      errorText,
+      helperText,
+      hideLabel,
+      invalid,
+      id,
+      label,
+      labelText,
+      onChange = () => null,
+      required,
+      style,
+      successText,
+      tooltipLabel,
+      tooltipText,
+      tooltipButtonLabel,
+      type = 'text',
+      ...rest
+    }: TextInputProps,
+    ref?: React.Ref<HTMLInputElement>,
+  ) => {
+    const defaultValues = getDefaultValues(defaultValue);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const hoursInputRef = useRef<HTMLInputElement>(null);
+    const minutesInputRef = useRef<HTMLInputElement>(null);
+    const [hours, setHours] = useState<string>(defaultValues ? defaultValues[0] : '');
+    const [minutes, setMinutes] = useState<string>(defaultValues ? defaultValues[1] : '');
+    const [time, setTime] = useState<string>(defaultValues ? defaultValues.join(':') : '');
+
+    const wrapperProps = {
+      className,
+      errorText,
+      helperText,
+      hideLabel,
+      id,
+      invalid,
+      label,
+      labelText,
+      required,
+      style,
+      successText,
+      tooltipLabel,
+      tooltipText,
+      tooltipButtonLabel,
+    };
+
+    /**
+     * Update the full time input and dispatch the native onChange event
+     */
+    const updateTimeInput = (newHours: string, newMinutes: string) => {
+      setHours(newHours);
+      setMinutes(newMinutes);
+      const newTimeValue = `${newHours}:${newMinutes}`;
+      setTime(newTimeValue);
+      const nativeInputValueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, 'value').set;
+      nativeInputValueSetter.call(inputRef.current, newTimeValue);
+      const event = new Event('input', { bubbles: true });
+      inputRef.current.dispatchEvent(event);
+    };
+
+    /**
+     * Merge props.ref to the internal ref
+     */
+    useLayoutEffect(() => {
+      if (ref) {
+        if (isFunction(ref)) {
+          (ref as (instance: HTMLInputElement) => void)(inputRef.current);
```

Comment:
What is happening here?

Reply:
I think it the hidden input is set as ref here. I think this can be a hook or a ref created with createRef.